contracts/cryptoblades.sol

```diff
@@ -543,22 +548,21 @@ contract CryptoBlades is Initializable, AccessControlUpgradeable {
                 .mul(ABDKMath64x64.fromUInt(chosenElementFee));
         _payContractStakedOnly(msg.sender, usdToSkill(discountedMintWeaponFee));
         _updateWeaponMintFee(quantity);
+        if (eventId > 0) {
+            specialWeaponsManager.addShards(msg.sender, eventId, quantity);
+        }
         SafeRandoms(links[LINK_SAFE_RANDOMS]).requestSingleSeed(msg.sender, getSeed(uint(WEAPON_SEED), quantity, chosenElement));
+        claimWeaponData[msg.sender] = uint256(uint256(chosenElement) | (uint256(quantity) << 32));
     }
 
-    function mintWeapon(uint32 quantity, uint8 chosenElement, uint256 eventId) external onlyNonContract oncePerBlock(msg.sender) {
-        require(quantity > 0 && quantity <= 10);
-        if(eventId > 0) {
-            specialWeaponsManager.addShards(msg.sender, eventId, quantity);
-        }
+    function mintWeapon() external onlyNonContract oncePerBlock(msg.sender) {
+        uint8 chosenElement = uint8((claimWeaponData[msg.sender]) & 0xFF);
+        uint32 quantity = uint32((claimWeaponData[msg.sender] >> 32) & 0xFFFFFFFF);
```

Comment:
I think we should require quantity > 0 here so the transaction fails with an error instead of passing and not minting anything while data is = 0.